pkg/metrics/exporter.go

```diff
@@ -41,7 +42,42 @@ func RegisterPrometheusExporter(registry *prom.Registry) (http.Handler, error) {
 		return nil, err
 	}
 	view.RegisterExporter(pe)
-	// since we're using prometheus we can report faster as we're only exposing metrics in memory
-	view.SetReportingPeriod(1 * time.Second)
+
 	return pe, nil
 }
+
+// RegisterStackdriverExporter register a Stackdriver exporter to OpenCensus.
+// It will add Agones metrics into Stackdriver on Google Cloud.
+func RegisterStackdriverExporter() error {
+	// Default project will be used
+	sd, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID: "",
+		// MetricPrefix helps uniquely identify your metrics.
+		MetricPrefix: "agones",
+	})
+	if err != nil {
+		return err
+	}
+	// It is imperative to invoke flush before your main function exits
+	defer sd.Flush()
```

Comment:
this defer should be part of the main function ! here it will be triggered right after the registration call but not before the main exist.